test/client_rebuild_test.go

```diff
@@ -438,3 +438,37 @@ func TestRebuildWithMissingMetadataDir(t *testing.T) {
 		"pack/a/2": {content: "pack/a/2 v1"},
 	})
 }
+
+func TestRebuildFileBecomesADir(t *testing.T) {
+	tc := util.NewTestCtx(t, auth.Project, 1)
+	defer tc.Close()
+
+	writeProject(tc, 1, 2)
+	writeObject(tc, 1, 1, i(2), "a.html", "a v1")
+	writeObject(tc, 1, 2, nil, "a.html/foo", "a v2")
```

Comment:
Could we do this instead?

```go
	writeObject(tc, 1, 1, i(2), "a.html", "a v1")
	writeObject(tc, 1, 2, nil, "a/b.html", "a v2")
```

This ☝️ feels like a more realistic example.

Reply:
That will not trigger the error. You need to turn a file into a directory of the same name

Reply:
ah I see! kk 👍